corehq/apps/sms/api.py

```diff
@@ -153,9 +154,14 @@ def send_sms(domain, contact, phone_number, text, metadata=None, logged_subevent
     if domain and contact and is_commcarecase(contact):
         backend_name = contact.get_case_property('contact_backend_id')
         backend_name = backend_name.strip() if isinstance(backend_name, str) else ''
+
         if backend_name:
             try:
                 backend = SQLMobileBackend.load_by_name(SQLMobileBackend.SMS, domain, backend_name)
+
+                if not _backend_enabled_for_domain(backend, domain):
```

Comment:
does this need to be applied to `send_sms_to_verified_number` and `send_sms_with_backend_name`?

Reply:
@snopoke You're right. Sloppy on my part. Same for [this case](https://github.com/dimagi/commcare-hq/pull/29892#discussion_r650819240)